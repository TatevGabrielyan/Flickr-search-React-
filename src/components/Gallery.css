.Gallery {
  display: flex;
  flex-wrap: wrap;
  margin: -1rem -1rem;
  padding-left: 15px;
  padding-right: 15px;
}
.animals{
  width: 300px;
  height: 85px;
  background-color: #C4D8E8;
  border-radius: 100px;
  border:none;
  color:white;
  font-weight: 800;
  text-align: center;
  font-size: 30px;
}
.dragAndDrop{
  margin-top: 5%;
}
.dropDiv{
  border:1px solid black;
  height: 200px;
}
@supports (display: grid) {
  .Gallery {
    display: grid;
    grid-template-columns: repeat(auto-fit, 150px);
    grid-gap: 2rem;
  }

  .Gallery {
    margin: 0;
  }
}
