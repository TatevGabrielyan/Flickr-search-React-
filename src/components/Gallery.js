import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Gallery.css';

import GalleryPagination from './GalleryPagination';
import GalleryItem from './GalleryItem';

import { setIdDetail } from '../actions';
import {
  getSearchIsLoading,
  getSearchPhotos,
  getSearchStat,
} from '../reducers';

class Gallery extends PureComponent {
  getGalleryItems = () => {
    const { photos, setIdDetail } = this.props;
    photos.length=5;
    if (!photos && !photos.lenght) return null;
    return photos.map(item => {
      const { id, title, url_q } = item;
      return (
        <GalleryItem
          key={id}
          id={id}
          title={title}
          url_q={url_q}
          setIdDetail={setIdDetail}
        />
      );
    });
  };

  render() {
    return (
      <Fragment>
        <GalleryPagination />
        <div className="Gallery">{this.getGalleryItems()}</div>
        <div className="dragAndDrop">
          <button className="animals">cat</button>
          <button className="animals">dog</button>
          <div className="dropDiv"> 
            
          </div>
        </div>
      </Fragment>
    );
  }
}

Gallery.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  photos: PropTypes.array.isRequired,
  stat: PropTypes.string,
  setIdDetail: PropTypes.func.isRequired,
};

const mapStateToProps = state => {
  return {
    isLoading: getSearchIsLoading(state),
    photos: getSearchPhotos(state),
    stat: getSearchStat(state),
  };
};

export default connect(
  mapStateToProps,
  { setIdDetail }
)(Gallery);
