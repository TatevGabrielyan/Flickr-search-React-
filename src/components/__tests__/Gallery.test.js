import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import Gallery from '../Gallery';
import store from '../../config/mockStore';

describe('Gallery component', () => {
  const component = (
    <Provider store={store}>
      <Gallery />
    </Provider>
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
