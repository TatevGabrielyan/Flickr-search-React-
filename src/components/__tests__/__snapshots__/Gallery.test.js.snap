// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component Snapshot matchs 1`] = `
Array [
  <div
    className="GalleryPagination"
  >
    <button
      className="GalleryPaginationButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
    <span
      className="GalleryPaginationText"
    >
      Page 
      5
       of 
      3710
       
    </span>
    <button
      className="GalleryPaginationButton"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>,
  <div
    className="Gallery"
  >
    <div
      className="GalleryItem"
    >
      <img
        alt="Großes Bügelbild GANGSTA CAT, Print Patch zum aufbügeln"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1927/44741364214_0f0a5195e9_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="that is a huge cat"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1956/45414777002_21af9f1855_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Spot in the Sun"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1948/30526186977_7dd1391bbd_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Indcar Iveco Wing - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1931/44741293554_5deb48dd5f_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Indcar Iveco Wing - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1931/30526171047_6e2d9250e4_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Indcar Iveco Wing - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1952/30526168187_fffebb861b_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Indcar Iveco Wing - Intérieur - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1974/43649301010_426016c0e1_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Red Panda (Ailurus fulgens)"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1946/44533196904_8172fa6ca1_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="พิธีฌาปนกิจศพ น.ส.สุดารัตน์ เชื่อมาก พยา บาลวิชาชีพ โรงพยาบาลส่งเสริมสุขภาพชุมชน ( รพ.สต.) เขาคอก เสียชีวิตขณะปฏิบัติหน้าที่เม ื่อวันที่ 15 ต.ค. 2561"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1943/44741246514_e48c396f79_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="_MG_1117_DxO"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1907/30526088107_2c7ddd584b_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Soul"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1974/45466393471_5096082038_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Nước hoa Tester Guerlain La Petite Robe Noire EDP 100ml"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1960/30526062937_f7310e3011_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Cat On The Tow Path"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1920/44741131414_bfce985903_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Dubrovnik"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1901/31591388548_2e4e379e34_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Dubrovnik"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1921/44741064614_be714bf4e2_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Dubrovnik"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1977/31591387098_6db6b56521_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Zadar, Croatia"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1961/44551893505_a3122ca23e_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1913/45414574872_d89aed9b64_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1934/44551955515_bfcfd22961_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1980/30525973157_2bc508e288_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1944/30525970897_d53090c3dc_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1931/45414560532_0f67d75edf_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1945/45414556722_1fa1c386af_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1970/44551935755_aca74a6dd9_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1959/44551931815_7565272c25_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1908/45466279571_562232274e_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1950/45466277891_52c15af3e0_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1926/43649121450_f0390902a0_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Myrtille"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1960/45414539372_972b1199f7_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1952/31591403158_4bb0d2447f_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1911/31591400008_f038fb2007_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1969/30525951367_7eb6523dea_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1932/44551910155_c682014d51_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1925/31591392648_fb5d2df745_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1966/44551902985_bc3b38fd2d_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1933/31591387438_caeea21b30_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1937/45466255731_a205ef2f4b_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Khuyến mãi SỮA TƯƠI TIỆT TRÙNG PROMESS HƯƠNG VANI 200ML LỐC 6 giá rẻ tại HAMYSHOP"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1910/44551892905_3156c9d437_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Crew Alta Festival"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1943/43649102250_969aeb3574_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="TRENDING NOW!! HOT 2017 Jewelry!!"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1902/30525934877_a50497a22a_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Irisbus Magelys HD - Poste de conduite - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1917/31591249068_61e1ef9fb3_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Irisbus Magelys HD - Intérieur - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1953/45414371032_2cdb7a2c06_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Irisbus Magelys HD - Intérieur - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1976/44740917294_d62ea4d169_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Irisbus Magelys HD - Intérieur - CAT 22"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1922/45414354222_49b9e7640d_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Healed Cat Portrait best Tattoo Black and Grey Realistic 3D by Jackie Rabbit"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1928/45465995771_be36c300ff_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Hugh Town aka Llareggub"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1914/44551785185_7ff480eed5_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Affût — Nanterre, octobre 2018"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1924/45414384152_013cf05790_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="1965-2018 Enlarge Saigon Execution Wall - Pháp trường cát 041018 38,16"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1909/31591173878_b5e085ae0e_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Tác giả ‘Thằng điên’ chia sẻ về vợ hot girl và những ngày thiếu tiền"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1906/30525733087_f3940c4e0d_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Queen and her bodyguard"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1972/45466041051_f359691d26_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Queen and her bodyguard"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1935/45466039441_704d6878fe_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Queen and her bodyguard"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1945/45466037991_a379c62ceb_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Chloe Moretz Photo Cat Morettz"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1952/44740774434_2150e8f058_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Chloe Moretz Photo Cat Moretzz"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1901/30525691377_ff76bda2bb_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="You Ready?"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1945/44740714784_3b7ddc58a9_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="P1530800 Golden Retriever Loan 241217"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1969/44740693984_ef571e6600_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="P1530799 Cocker Spaniel 3"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1923/43648758370_11261d8a8c_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="P1530801 Rottweiler"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1950/43648757810_fe42ca1532_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="180927 River Seine Cruise 34 Givenny Monets cat"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1925/44551339265_786174e006_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Gypsy. One of the Lodge cats @for_ladies_who_lodge"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1954/44551530555_ffcb1ec5ba_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Canon_EOS_5Ds_Rizik_my_action_photos_cat_06"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1912/44551470475_73e9d5226f_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="Windy Sunday"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1944/44740496594_9ef9d91900_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181020-DSCF4069-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1961/31590872268_3cd39bfd4f_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181020-DSCF4074-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1936/43648570460_954a691054_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4109-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1910/31590871808_6f0ec8790b_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4097-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1940/30525420747_75492624b1_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4139-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1953/30525420397_758283e654_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4135-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1932/43648569720_faf4727a7c_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4154-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1957/44740480334_54841f2a71_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4179-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1971/31590871038_362b6f7428_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4157-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1954/30525419787_81b2d6ba91_q.jpg"
      />
    </div>
    <div
      className="GalleryItem"
    >
      <img
        alt="20181021-DSCF4162-2"
        className="GalleryImage"
        onClick={[Function]}
        src="https://farm2.staticflickr.com/1963/43648569140_82bceb0b4c_q.jpg"
      />
    </div>
  </div>,
]
`;
